### PR TITLE
Nü Note Options: Mark IV

### LINKED ISSUE #763: Spring Cleaning: Nü Note Options
### Guidelines
- New code should always be Swift
- Autolayout **is always preferred**
- Please drop outdated components when appropriate! 🔥🔥🔥
- [Simplenote Icons Here](https://github.com/Automattic/Simplecons)

### Details:
- [x] Implement a new Note Options component, and replace the legacy UI

### PR DESCRIPTION
**Pull request by jleandroperez**
### 🛑  Important
Don't merge until #906 #907 #909 and #910 are ready: everything must be merged in a single shot!

### Details:
In this PR we're wiring support for few missing Note Option Actions:

- Share
- History
- Trash

@aerych Sir!! can I trouble you with this one?
Thanks in advance!!

Ref. #763

### Test
1. Log into your account
2. Open any note
3. Press over the top `( i )` button

- [x] Verify pressing **Share** dismisses the Options and presents the Share Sheet
- [x] Verify pressing **History** dismisses the Options and presents the History Slider
- [x] Verify pressing **Trash** dismisses both, Options and the Editor, and trashes the Note
- [x] Verify the same works super on an iPad device please!!

### Release
These changes do not require release notes.

### CODE REVIEW

File: Simplenote/Classes/SPNoteEditorViewController+Extensions.swift [hunk 1]
```diff
@@ -224,6 +226,62 @@ private extension SPNoteEditorViewController {
...
+            self.presentHistoryController()
+        }
+    }
+
+    func optionsControllerDidPressTrash(_ sender: OptionsViewController) {
+        sender.dismiss(animated: true, completion: nil)
+
+        // Wait a bit until the Dismiss Animation concludes. `dismiss(:completion)` takes too long!
+        DispatchQueue.main.asyncAfter(deadline: .now() + UIKitConstants.animationDelayShort) {
+            self.delete(note: sender.note)
```
Comment (aerych) [inline Simplenote/Classes/SPNoteEditorViewController+Extensions.swift:282]: When trashing a note we want to dismiss the editor, yeah?  Are we missing a call to `self.backButtonAction(sender)` here?

**Review by aerych — CHANGES_REQUESTED** [review]
Hola @jleandroperez 👋 
Code changes look good over all.  I just had two comments (only the first one needs a change):
1. The editor isn't dismissing when trashing.  Looks like we never call `backButtonAction`? Maybe a commit that didn't get pushed?
2. I can dismiss the share sheet by tapping where the screen is greyed out. This trained me to try to do the same with the history slider but I have to tap the Cancel button there.  Would be a nice future enhancement maybe. :) 

Looks good to ship once the editor dismisses! :)

**Comment by jleandroperez** [comment]
Thank you @aerych!! The `back` invocation was there, at some point, buuut i've been moving code between branches (just trying to find a good balance!), and it got lost along the way.

History Slider is already redone elsewhere (Evgeny's trial project!). 

Thanks a loooot!!

**Review by aerych — APPROVED** [review]

**Comment by aerych** [comment]
:shipit: sir!!

**Comment by jleandroperez** [comment]
Thank you Eric!!

**Pull request merged**